The `getString()` function of `ShardingSphereResultSet` should try to convert the object before throwing an exception

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

- 5.2.1

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

- ShardingSphere-JDBC

### Expected behavior

- SQL for logical tables is executed normally.

### Actual behavior

- SQL for logical tables isnot executed normally.
```shell
Caused by: java.lang.ClassCastException: class java.time.LocalDateTime cannot be cast to class java.lang.String (java.time.LocalDateTime and java.lang.String are in module java.base of loader 'bootstrap')
	at org.apache.shardingsphere.driver.jdbc.core.resultset.ShardingSphereResultSet.getString(ShardingSphereResultSet.java:160)
	at com.zaxxer.hikari.pool.HikariProxyResultSet.getString(HikariProxyResultSet.java)
	at org.apache.ibatis.logging.jdbc.ResultSetLogger.printColumnValues(ResultSetLogger.java:111)
	at org.apache.ibatis.logging.jdbc.ResultSetLogger.invoke(ResultSetLogger.java:80)
	at jdk.proxy4/jdk.proxy4.$Proxy132.next(Unknown Source)
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleRowValuesForSimpleResultMap(DefaultResultSetHandler.java:359)
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleRowValues(DefaultResultSetHandler.java:335)
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleResultSet(DefaultResultSetHandler.java:308)
	at org.apache.ibatis.executor.resultset.DefaultResultSetHandler.handleResultSets(DefaultResultSetHandler.java:201)
	at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:65)
	at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:79)
	at org.apache.ibatis.executor.SimpleExecutor.doQuery(SimpleExecutor.java:63)
	at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:325)
	at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:156)
	at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:109)
	at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:89)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:151)

```

### Reason analyze (If you can)

- In the SpringBoot application, configuring `mybatis.configuration.log-impl` to `org.apache.ibatis.logging.stdout.StdOutImpl` will allow users to see the SQL Log. Correspondingly, in the `org.apache.ibatis.logging.jdbc.ResultSetLogger#printColumnValues` function, it will directly call the `getString` method of `java.sql.ResultSet`.
- ![image](https://user-images.githubusercontent.com/20187731/206276047-0ed0886b-53b2-47dc-a402-c6b0892bf60b.png)

- When ShardingSphere operates the real table, there is no problem in this step. But once the logical table of ShardingSphere JDBC is operated, `java.sql.ResultSet` is actually `org.apache.shardingsphere.driver.jdbc.core.resultset.ShardingSphereResultSet`, and its `getString` function will try to convert `java.time.LocalDateTime `Similar classes convert to `java.lang.String`, which is not possible to do directly.
- ![image](https://user-images.githubusercontent.com/20187731/206276004-05ebccd7-3de6-4acc-9251-7783696268f5.png)

- Apparently, removing the configuration of `mybatis.configuration.log-impl` will solve this problem, but this should not be a reasonable option.
- ![image](https://user-images.githubusercontent.com/20187731/206278336-3cb4b473-2631-45f0-a0cb-8845eb4f944c.png)


### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

- After ensuring Docker CLI is installed, execute the following command. Git at https://github.com/linghengqian/mybatis-stdout-shardingsphere-test .
- The logical table test of the unit test fails, the real table test is fine.

```shell
git clone git@github.com:linghengqian/mybatis-stdout-shardingsphere-test.git
cd ./mybatis-stdout-shardingsphere-test/
sdk install java 22.3.r17-grl
sdk use java 22.3.r17-grl
./gradlew clean test
```

### Example codes for reproduce this issue (such as a github link).

- Git at https://github.com/linghengqian/mybatis-stdout-shardingsphere-test
- This is the real reason for https://github.com/apache/shardingsphere/issues/22451 .

--- a/infra/executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/util/ResultSetUtil.java
+++ b/infra/executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/util/ResultSetUtil.java
@@ -86,7 +86,11 @@ public final class ResultSetUtil {
         if (String.class.equals(convertType)) {
             return value.toString();
         }
-        throw new SQLFeatureNotSupportedException("getObject with type");
+        try {
+            return convertType.cast(value);
+        } catch (ClassCastException cce) {
+            throw new SQLFeatureNotSupportedException("getObject with type");
+        }
     }
     
     private static Object convertURL(final Object value) {
@@ -135,6 +139,9 @@ public final class ResultSetUtil {
         if (Timestamp.class.equals(convertType)) {
             return Timestamp.valueOf(localDateTime);
         }
+        if (String.class.equals(convertType)) {
+            return value.toString();
+        }
         return value;
     }
     
@@ -151,6 +158,9 @@ public final class ResultSetUtil {
         }
         if (OffsetDateTime.class.equals(convertType)) {
             return timestamp.toInstant().atZone(ZoneId.systemDefault()).toOffsetDateTime();
+        }
+        if (String.class.equals(convertType)) {
+            return value.toString();
         }
         return value;
     }

--- a/jdbc/core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSetTest.java
+++ b/jdbc/core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ShardingSphereResultSetTest.java
@@ -42,7 +42,6 @@ import java.sql.Ref;
 import java.sql.ResultSet;
 import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
-import java.sql.SQLFeatureNotSupportedException;
 import java.sql.SQLXML;
 import java.sql.Time;
 import java.sql.Timestamp;
@@ -55,8 +54,8 @@ import java.util.Properties;
 
 import static org.hamcrest.CoreMatchers.instanceOf;
 import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertFalse;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.mock;
@@ -207,7 +206,12 @@ public final class ShardingSphereResultSetTest {
     @Test
     public void assertGetStringWithColumnIndex() throws SQLException {
         when(mergeResultSet.getValue(1, String.class)).thenReturn("value");
+        LocalDateTime tempTime = LocalDateTime.of(2022, 12, 14, 0, 0);
+        when(mergeResultSet.getValue(2, String.class)).thenReturn(tempTime);
+        when(mergeResultSet.getValue(3, String.class)).thenReturn(Timestamp.valueOf(tempTime));
         assertThat(shardingSphereResultSet.getString(1), is("value"));
+        assertThat(shardingSphereResultSet.getString(2), is("2022-12-14T00:00"));
+        assertThat(shardingSphereResultSet.getString(3), is("2022-12-14 00:00:00.0"));
     }
     
     @Test
@@ -616,11 +620,11 @@ public final class ShardingSphereResultSetTest {
         assertThat(shardingSphereResultSet.getObject(1, Clob.class), is(result));
     }
     
-    @Test(expected = SQLFeatureNotSupportedException.class)
+    @Test
     public void assertGetObjectWithRef() throws SQLException {
         Ref result = mock(Ref.class);
         when(mergeResultSet.getValue(1, Ref.class)).thenReturn(result);
-        shardingSphereResultSet.getObject(1, Ref.class);
+        assertThat(shardingSphereResultSet.getObject(1, Ref.class), is(result));
     }
     
     @Test

